common/djangoapps/third_party_auth/utils.py

```diff
@@ -24,6 +33,25 @@ def user_exists(details):
         user_queryset_filter['username'] = username
 
     if user_queryset_filter:
-        return User.objects.filter(**user_queryset_filter).exists()
+        if settings.FEATURES.get('APPSEMBLER_MULTI_TENANT_EMAILS', False):
+            current_org = get_current_organization()
+
+            if email:
+                return current_org.userorganizationmapping_set.filter(user__email=email).exists()
+            elif username:
+                try:
+                    user = User.objects.get(username=username)
+                except User.DoesNotExist:
+                    return False
+
+                if get_single_user_organization(user) == current_org:
+                    return True
+                else:
+                    raise SAMLUnusableUsernameDueToMTE(
```

Comment:
Logging to add more info.

```suggestion
                    log.info('SAMLUnusableUsernameDueToMTE: Cannot use username (%s) in (%s) because it is already taken in a different organization (%s)', username, current_org.short_name, user_org.short_name)
                    raise SAMLUnusableUsernameDueToMTE(
```